tests/test_Exclusivity.py

```diff
@@ -1,78 +1,137 @@
+    ExclusivityContract.addToTopicsList(accounts[1])
 
-def testAddNonTopics(ExclusivityContract, accounts):
+    # test above 0.5 - should remove
+    ExclusivityContract.setPercentage(51)
+    assert ExclusivityContract.tallyVotes(accounts[0]).return_value
 
-    ExclusivityContract.addToTopicsList(accounts[2])
-    ExclusivityContract.setPercentage(100)#SET TO 20 FOR TEST TESTING, CHANGE TO 100 LATER
+    # test below 0.5 - shouldn't remove
+    ExclusivityContract.setPercentage(40)
+    assert not ExclusivityContract.tallyVotes(accounts[1]).return_value
+
+def testAddNonTopics(ExclusivityContract, accounts):
+    # test successfully add case 
+    ExclusivityContract.setPercentage(100)
     ExclusivityContract.addNonTopics(accounts[1])
 
-    assert ExclusivityContract.isInTopicsList(accounts[1]).return_value,"should add if percentage is 100 or greater"
+    assert ExclusivityContract.isInTopicsList(accounts[1]).return_value, "should add if percentage is 100 or greater"
 
-    ExclusivityContract.popTopicList()
+    # test fail to add case
     ExclusivityContract.setPercentage(20)
-    ExclusivityContract.addNonTopics(accounts[1])
+    ExclusivityContract.addNonTopics(accounts[2])
     
-    assert not ExclusivityContract.isInTopicsList(accounts[1]).return_value,"should not add if percentage is lower than 100"
+    assert not ExclusivityContract.isInTopicsList(accounts[2]).return_value, "should not add if percentage is lower than 100"
 
 
-def testRemoveTopics(ExclusivityContract,accounts):
+def testRemoveTopics(ExclusivityContract, accounts):
 
     ExclusivityContract.addToTopicsList(accounts[2])
     ExclusivityContract.addToTopicsList(accounts[1])
     ExclusivityContract.addToTopicsList(accounts[3])
     ExclusivityContract.addToTopicsList(accounts[4])
 
+    # f = open("account_test.txt", "w") 
+    # for adr in ExclusivityContract.getTopicsList().return_value:
+    #     f.write(adr)
+    #     f.write("\n")
+    # f.write("\n trying remove 1")
+
     ExclusivityContract.setPercentage(100)
-    ExclusivityContract.removeNonTopics(accounts[1])
+
+    removed_success: bool = ExclusivityContract.removeTopics(accounts[1]).return_value
     
-    assert not ExclusivityContract.isInTopicsList(accounts[1]).return_value, "should remove if percentage is greater than or equal to 100"
-    ExclusivityContract.addToTopicsList(accounts[1])
+    assert removed_success, "should remove if percentage is greater than or equal to 100"
+
+    removed_success = False
+
+    # for adr in ExclusivityContract.getTopicsList().return_value:
+    #     f.write(adr)
+    #     f.write("\n")
+    # f.write("\n trying remove 2")
 
     ExclusivityContract.setPercentage(2)
-    ExclusivityContract.removeNonTopics(accounts[1])
-    assert ExclusivityContract.isInTopicsList(accounts[1]).return_value, "should not remove if percentage is less than 100"
 
+    removed_success = ExclusivityContract.removeTopics(accounts[2]).return_value
```

Comment:
Once again, ActiveUser already has the structure to hold this and should be contract which is called.  In order to do that, this contract should be added as an Admin in the ActiveUser contract on __init__